packages/http/src/validator/validators/security/handlers/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { DiagnosticSeverity } from '@stoplight/types';
+import { Either, left, right } from 'fp-ts/lib/Either';
+import { IPrismDiagnostic } from '@stoplight/prism-core';
+import { IHttpRequest } from '../../../../types';
+
+export type ValidateSecurityFn = (
+  input: Pick<IHttpRequest, 'headers' | 'url'>,
+  name: string
+) => Either<IPrismDiagnostic, boolean>;
+
+export function genRespForScheme(
+  isSchemeProper: boolean,
+  isCredsGiven: boolean,
+  msg: string
+): Either<IPrismDiagnostic, boolean> {
+  if (isSchemeProper) {
+    return when(isCredsGiven, undefined);
+  }
+
+  return left(genUnauthorisedErr(msg));
+}
+
+export const genUnauthorisedErr = (msg?: string): IPrismDiagnostic => ({
+  severity: DiagnosticSeverity.Error,
+  message: 'Invalid security scheme used',
+  code: 401,
+  tags: msg ? [msg] : [],
+});
+
+export function isScheme(shouldBeScheme: string, authScheme: string) {
+  return authScheme.toLowerCase() === shouldBeScheme;
+}
+
+export function when(condition: boolean, errorMessage: string | undefined): Either<IPrismDiagnostic, boolean> {
```

Comment:
`boolean` seems to not be used anywhere, maybe `unknown` would be better?

Reply:
I'm not too sure I understand. The condition is evaluated in the following line

Reply:
Shit, wrong line.

Reply:
I was referring to `boolean` in `Either<IPrismDiagnostic, boolean>`. I missed that `boolean` is also type of condition arg.